server/src/main/java/org/elasticsearch/gateway/GatewayService.java

```diff
@@ -221,11 +214,7 @@ public ClusterState execute(final ClusterState currentState) {
                 logger.debug("cluster is already recovered");
                 return currentState;
             }
-
-            return Function.<ClusterState>identity()
-                .andThen(ClusterStateUpdaters::updateRoutingTable)
-                .andThen(ClusterStateUpdaters::removeStateNotRecoveredBlock)
-                .apply(currentState);
+            return ClusterStateUpdaters.removeStateNotRecoveredBlock(ClusterStateUpdaters.updateRoutingTable(currentState));
```

Comment:
Unrelated but this was just the weirdest way ever to call two functions ...

Reply:
It's pretty weird that this is even two functions, this is the only place they're called (apart from `ClusterStateUpdatersTests` which really needn't test the two things in isolation either). I've nearly opened a PR to fix this a number of times myself, thanks.